runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -249,13 +275,12 @@ private void issueBatchedRequest(Map<Long, PendingRequest> requests) {
   }
 
   private void issueMultiChunkRequest(final long id, PendingRequest pendingRequest) {
-    checkNotNull(pendingRequest.computation);
-    final ByteString serializedCommit = pendingRequest.request.toByteString();
-
+    checkNotNull(pendingRequest.computationId());
+    final ByteString serializedCommit = pendingRequest.serializedCommit();
     synchronized (this) {
       pending.put(id, pendingRequest);
       for (int i = 0;
-          i < serializedCommit.size();
+          i < serializedCommit.size() && !isShutdown();
```

Comment:
I'd remove the shutdown check since it's not consistent with other paths and probably won't matter if the send just won't do anything anyway.

Reply:
done